src/xwayland/xserver.rs

```diff
@@ -80,37 +76,46 @@ pub struct XWayland<Data> {
 /// disconnecting several time in a row, but only a single connection will
 /// be active at any given time.
 #[derive(Debug)]
+#[must_use = "Connection events must be handled to prevent fd leaking"]
 pub enum XWaylandEvent {
     /// The XWayland server is ready
     Ready {
         /// Privileged X11 connection to XWayland
         connection: UnixStream,
+
         /// Wayland client representing XWayland
-        client: Client,
+        ///
+        /// This should be used to create a Wayland client using [`Display::insert_client`](wayland_server::Display::insert_client).
+        ///
+        /// When the client is disconnected, you should [`shutdown`](XWayland::shutdown) XWayland.
+        client: UnixStream,
+
+        /// The display number the XWayland server is available at.
+        ///
+        /// This may be used if you wish to set the `DISPLAY` variable manually when spawning processes that
+        /// may use XWayland.
```

Comment:
Should this comment mention something like `format!(":{}", event.display)`? That might be helpful in figuring out how to set `$DISPLAY`.

Reply:
Yes, I should note that.